src/test/java/tools/jackson/databind/deser/enums/EnumNamingDeserializationTest.java

```diff
@@ -291,4 +291,17 @@ public void testEnumMixInDeserializationTest() throws Exception {
         BaseEnum deser = mapper.readValue(q("realName"), BaseEnum.class);
         assertEquals(BaseEnum.REAL_NAME, deser);
     }
+
+    @Test
+    void testUseEnumMappingStrategySetInMapper() {
+        ObjectMapper mapper = jsonMapperBuilder()
+                .enumNamingStrategy(EnumNamingStrategies.CamelCaseStrategy.INSTANCE)
```

Comment:
This seems awkward.
`.enumNamingStrategy(EnumNamingStrategies.CamelCaseStrategy.class)` then handle the instance internally?

Reply:
No, I think strategy passed should be instance, not class. This is used for most configuration; mapper/builder not having to dynamically construct instances.
If we want instantiation, should instead pass factory; `Class` only to be used in cases where we must (from Annotation where you cannot indicate instance but only Class).

Reply:
We could just ask for `new EnumNamingStrategies.CamelCaseStrategy()`, but if there's already `INSTANCE` can use that.

Reply:
Ah, makes sense👍🏼